src/test/scala/mesosphere/marathon/state/TaskKillerTest.scala

```diff
@@ -0,0 +1,74 @@
+package mesosphere.marathon.state
+
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.tasks.TaskTracker
+import mesosphere.marathon.upgrade.DeploymentPlan
+import mesosphere.marathon.{ MarathonSchedulerService, MarathonSpec }
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ BeforeAndAfterAll, Matchers }
+import org.mockito.Mockito._
+import org.mockito.Matchers.{ any, anyBoolean }
+
+import scala.concurrent.Future
+
+class TaskKillerTest extends MarathonSpec
+    with Matchers
+    with BeforeAndAfterAll
+    with MockitoSugar
+    with ScalaFutures {
+
+  var tracker: TaskTracker = _
+  var service: MarathonSchedulerService = _
+  var groupManager: GroupManager = _
+  var taskKiller: TaskKiller = _
+
+  before {
+    service = mock[MarathonSchedulerService]
+    tracker = mock[TaskTracker]
+    groupManager = mock[GroupManager]
+    taskKiller = new TaskKiller(tracker, groupManager, service)
+  }
+
+  test("AppNotFound") {
+    val appId = PathId("invalid")
+    when(tracker.contains(appId)).thenReturn(false)
+
+    val result = taskKiller.kill(appId, (tasks) => Set.empty[MarathonTask], scale = true)
+    result.futureValue shouldEqual TaskKiller.AppNotFound(appId)
+  }
+
+  test("GroupNotFound") {
+    val appId = PathId(List("my", "app"))
+    when(tracker.contains(appId)).thenReturn(true)
+    when(groupManager.group(appId.parent)).thenReturn(Future.successful(None))
+
+    val result = taskKiller.kill(appId, (tasks) => Set.empty[MarathonTask], scale = true)
+    result.futureValue shouldEqual TaskKiller.GroupNotFound(appId.parent)
+  }
+
+  test("KillRequested") {
+    val appId = PathId(List("my", "app"))
+    val tasksToKill = Set(MarathonTask.getDefaultInstance)
+    when(tracker.contains(appId)).thenReturn(true)
+    when(groupManager.group(appId.parent)).thenReturn(Future.successful(Some(Group.emptyWithId(appId.parent))))
+    when(groupManager.updateApp(any[PathId], any(), any[Timestamp](), anyBoolean(), any[Set[MarathonTask]]())).thenReturn(
```

Comment:
Please make you test more specific here. You should test that the correct parameters are passed to updateApp. E.g. by capturing them and then test them. This is really the good case and it should be tested completely.